### ISSUE TITLE
Regression: 1.0 can't deserialize to non-public records

### ISSUE THREAD

**Issue by cmeeren** [Tarmil/FSharp.SystemTextJson#133]
I haven't done extensive testing, so the problem may be bigger than just internal records. But this fails with an exception:

```f#
open System.Text.Json
open System.Text.Json.Serialization

type internal T = { foo: string }
let json = """{ "foo": "" }"""

let private options = JsonSerializerOptions()
options.Converters.Add(JsonFSharpConverter())

JsonSerializer.Deserialize<T>(json, options) |> ignore
```

If I remove `internal` from the type definition, it works.

This worked fine in 0.19.13.

**Comment by Tarmil** [comment]
Fixed in v1.0.6 🙂

**Comment by cmeeren** [comment]
Excellent, thank you so much! 🎉
